consensus/src/dag/commit_signer.rs

```diff
@@ -0,0 +1,28 @@
+use crate::experimental::signing_phase::CommitSignerProvider;
+use aptos_crypto::bls12381;
+use aptos_types::validator_signer::ValidatorSigner;
+
+pub struct DagCommitSigner {
+    signer: ValidatorSigner,
+}
+
+impl DagCommitSigner {
+    pub fn new(signer: ValidatorSigner) -> Self {
+        Self { signer }
+    }
+}
+
+impl CommitSignerProvider for DagCommitSigner {
+    fn sign_commit_vote(
+        &self,
+        _ledger_info: aptos_types::ledger_info::LedgerInfoWithSignatures,
+        new_ledger_info: aptos_types::ledger_info::LedgerInfo,
+    ) -> Result<bls12381::Signature, aptos_safety_rules::Error> {
+        let signature = self
+            .signer
+            .sign(&new_ledger_info)
+            .map_err(|err| aptos_safety_rules::Error::SerializationError(err.to_string()))?;
```

Comment:
We need to delete aptos_safety_rules. Can we move the error somewhere else?

Reply:
I think so. If and when we actually delete it, we can move?

Reply:
Sure.